### DISCUSSION TITLE
Changing the root div

### DISCUSSION THREAD

**Discussion by daraclare** [vercel/next.js#60254] [Help]
### Summary

My team was planning on using Nextjs for a project which is a library to be imported into multiple apps using a script tag (framework agnostic). The script then renders an ad (for internal use mostly). 
We planned to use the root div ID and change it to something specific for each case, so my Nextjs project would render in `<div id="myad-111" />` in one app and `<div id="myad-222" />` in another. 
However, I realise now that since Nextjs 13 the root div ID has been removed. What is the best way to accomplish this without reverting to Nextjs 12 or React? I'd appreciate any suggestions.

**Comment by icyJoseph** [comment]
Maybe you can add your own target div? In the root layout `app/layout.tsx`.

**Reply by daraclare** [reply]
Apologies, I didn't receive a notification. Thanks for replying! Do you mean by using createRoot in Nextjs? Will that work in Next v13 or v14 now that Nextjs no longer uses the `__next` id to target the root container? I was under the impression it wouldn't but maybe I'm wrong? Or do you mean by creating multiple portals?

**Comment by don-esteban** [comment]
Hi @daraclare, I was not eve aware of such a root `<div>`. I use latest Next.js and define the complete HTML structure in the root layout. Something like this:

```
export default function RootLayout({ children }: { children: React.ReactNode }) {
  return (
    <html lang="de" className={inter.variable + " scroll-smooth"}>
      <body className="flex min-h-screen flex-col">
        <Header />
        <main className="flex-grow"><div id="123">{children}</div></main>
        <Footer />
      </body>
    </html>
  );
}
```

Inside `<main>` I added a `div` with an `id`. Is it that, what you mean by root div? I can also add `id` attributes elsewhere, all shown up nicely in the resulting HTML.

**Reply by icyJoseph** [reply]
@daraclare this is what I mean, since in App Router, you are in full control of the html layout, you can just put your own div, for example `<div id="123">{children}</div>`, and then all pages will be put in place of children. Which mimics the Pages Router behavior you used to have.

**Reply by daraclare** [reply]
Thanks both, I'm looking for a more dynamic solution as I want to dynamically render multiple divs each with their own content. We've done a lot of research on our end and have come to the conclusion that Nextjs is overkill for what we need and have opted for a simple react app using Vite. Thank you both so much for the input, and apologies for the delay in replying, I am not getting notifications even though I'm subscribed, not sure why. Thanks again!

**Reply by neveenatik** [reply]
@daraclare trying to apply the same solution I just read in the docs there can be layout on each "group" level so for the use of library on each exported component https://nextjs.org/docs/app/building-your-application/routing/route-groups

**Comment by guugg** [comment]
I'm also looking for a'non-node.js' way to render'next.js widgets' (SPA), so far none. Currently also choose'Vite '+' react '.' next.js' can be'Static Exports', such exports generate many '< script >' in html
